components/infobox/extensions/commons/player_introduction.lua

```diff
@@ -0,0 +1,597 @@
+			.. ')'
+	end
+
+	return nameDisplay
+end
+
+--- builds the born display
+---@param isDeceased boolean
+---@return string
+function PlayerIntroduction:_bornDisplay(isDeceased)
+	if self.playerInfo.birthDate == DEFAULT_DATE then
+		return ''
+	end
+
+	if not isDeceased then
+		return ' (born '
+---@diagnostic disable-next-line: param-type-mismatch
+			.. os.date("!%B %d, %Y", tonumber(mw.getContentLanguage():formatDate('U', self.playerInfo.birthDate))):gsub(' 0',' ')
+			.. ')'
+	elseif self.playerInfo.deathDate ~= DEFAULT_DATE then
+		return ' ('
+---@diagnostic disable-next-line: param-type-mismatch
+			.. os.date("!%B %d, %Y", tonumber(mw.getContentLanguage():formatDate('U', self.playerInfo.birthDate))):gsub(' 0',' ')
+			.. ' – '
+---@diagnostic disable-next-line: param-type-mismatch
+			.. os.date("!%B %d, %Y", tonumber(mw.getContentLanguage():formatDate('U', self.playerInfo.birthDate))):gsub(' 0',' ')
+			.. ')'
+	end
+
+	return ''
+end
+
+--- builds the status display
+---@param isDeceased boolean
+---@return string?
+function PlayerIntroduction:_statusDisplay(isDeceased)
+	if self.playerInfo.status ~= 'active' and not isDeceased then
+		return self.playerInfo.status
+	end
+
+	return nil
+end
+
+--- builds the nationality display
+---@return string?
+function PlayerIntroduction:_nationalityDisplay()
+	local nationalities = {}
+	for _, nationality in Table.iter.pairsByPrefix(self.playerInfo, 'nationality', {requireIndex = false}) do
+		table.insert(nationalities, '[[:Category:' .. nationality:gsub("^%l", string.upper)
+				.. '|' .. (Flags.getLocalisation(nationality) or '') .. ']]')
+	end
+
+	if Table.isEmpty(nationalities) then
+		return nil
+	end
+
+	return table.concat(nationalities, '/')
+end
+
+--- builds the game display
+---@return string?
+function PlayerIntroduction:_gameDisplay()
+	if String.isEmpty(self.playerInfo.game) then
+		return nil
+	end
+
+	return self._addConcatText('<i>' .. self.playerInfo.game .. '</i>')
+end
+
+--- builds the faction display
+---@return string?
+function PlayerIntroduction:_factionDisplay()
+	if not self.options.showFaction or String.isEmpty(self.playerInfo.faction) or self.playerInfo.type ~= TYPE_PLAYER then
+		return nil
+	end
+
+	local factions = {}
+	for _, faction in Table.iter.pairsByPrefix(self.playerInfo, 'faction', {requireIndex = false}) do
+		table.insert(factions, '[[:Category:' .. faction .. '|' .. faction .. ']]')
+	end
+
+	return self._addConcatText(mw.text.listToText(factions, ', ', ' and '))
+end
+
+--- builds the type display
+---@return string?
+function PlayerIntroduction:_typeDisplay()
+	return self._addConcatText(self.playerInfo.type)
+		.. self._addConcatText(
+			self.playerInfo.type ~= TYPE_PLAYER and String.isNotEmpty(self.playerInfo.role2) and self.playerInfo.role2,
+		' and ')
+end
+
+--- builds the team and role display
+---@param isDeceased boolean
+---@return string?
+function PlayerIntroduction:_teamDisplay(isDeceased)
```

Comment:
this function can use some interpolation instead of nested if/else concats

Reply:
sadly it gets just as messy with interpolation

Reply:
pls check if you like it better now

(still needs testing though ...)

Reply:
Looks lot better! Some of the params with complicated logic could be set in a variable instead to make it look a bit mesy